cmd/minishift/cmd/oc_env.go

```diff
@@ -83,7 +90,14 @@ var ocEnvCmd = &cobra.Command{
 
 		var shellCfg *OcShellConfig
 
-		shellCfg, err = getOcShellConfig(config.InstanceConfig.OcPath, forceShell)
+		// When oc-env --profile PROFILE_NAME is used and PROFILE_NAME is not an active profile
+		var ocPath string
+		if constants.ProfileName != profileActions.GetActiveProfile() {
+			ocPath = getOcPath()
+		} else {
```

Comment:
Unnecessary else
```
ocPath = config.InstanceConfig.OcPath
if constants.ProfileName != profileActions.GetActiveProfile() {
    ocPath = getOcPath()
}
```

Reply:
@budhrg I do not like this because we are now overwriting the variable `ocPath` when we meet the if condition.

Reply:
In this situation, it is not the same as the return-else case. I am fine with this...
I am OK with this... it is a possible extra instruction, but this might be optimized away anyhow.